modules/spring-config/src/main/java/org/mule/runtime/config/internal/LazyMuleArtifactContext.java

```diff
@@ -201,11 +202,25 @@ protected void prepareBeanFactory(ConfigurableListableBeanFactory beanFactory) {
     addBeanPostProcessors(beanFactory, trackingPostProcessor);
   }
 
-  private void applyLifecycle(List<String> createdComponentModels, boolean applyStartPhase) {
+  private void applyLifecycle(Map<String, Object> components, boolean applyStartPhase) {
     muleContext.withLifecycleLock(() -> {
+      // Sorter in order to initialize and start the components according to they dependencies
+      ComponentConfigurationLifecycleObjectSorter componentConfigurationLifecycleObjectSorter =
+          new ComponentConfigurationLifecycleObjectSorter(beanName -> getDependencyResolver()
+              .resolveComponentDependencies(beanName).stream()
+              .map(componentName -> components.get(componentName))
+              .filter(component -> component != null)
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Replace this lambda with method reference 'Objects::nonNull'. ([squid:S1612](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AS1612))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule&pullRequest=7761&issues=AWnZSM1wotIXnmsU458_&open=AWnZSM1wotIXnmsU458_)

[ISSUE_KEY]: AWnZSM1wotIXnmsU458_